lib/Doctrine/DBAL/Platforms/AbstractPlatform.php

```diff
@@ -850,17 +850,15 @@ public function getLowerExpression($str)
     }
 
     /**
-     * Returns the SQL snippet to get the position of the first occurrence of substring $substr in string $str.
+     * Returns the SQL snippet to get the position of the first occurrence of the substring in the string.
      *
-     * @param string    $str      Literal string.
-     * @param string    $substr   Literal string to find.
-     * @param int|false $startPos Position to start at, beginning of string by default.
-     *
-     * @return string
+     * @param string      $string    String to locate the substring in, SQL expression.
+     * @param string      $substring Substring to locate, SQL expression.
+     * @param string|null $start     Position to start at, SQL expression. Defaults to the beginning of the string.
```

Comment:
Alao confusing - why is this a `string`? Could we maybe add example usages?

Reply:
Because it's an SQL expression, not a value (see the test). It could a reference to another column in the query or a function call or whatever else. There will be a lot of changes like this in the API. Not sure if we want to put an example in every occurrence. What would you suggest?